src/Uno.UWP/Devices/Sensors/Magnetometer.iOS.cs

```diff
@@ -0,0 +1,63 @@
+﻿#if __IOS__
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using CoreMotion;
+using Foundation;
+using Uno.Devices.Sensors.Helpers;
+
+namespace Windows.Devices.Sensors
+{
+	public partial class Magnetometer
+	{
+		private readonly CMMotionManager _motionManager;
+
+		private Magnetometer(CMMotionManager motionManager)
+		{
+			_motionManager = motionManager;
+		}
+
+		public uint ReportInterval
+		{
+			get => (uint)_motionManager.MagnetometerUpdateInterval * 1000;
+			set => _motionManager.MagnetometerUpdateInterval = value / 1000.0;
+		}
+
+		private static Magnetometer TryCreateInstance()
+		{
+			var motionManager = new CMMotionManager();
+			return !motionManager.MagnetometerAvailable ?
```

Comment:
A good practice is to put the positive result first, it's easier to read :)
```suggestion
			return motionManager.MagnetometerAvailable ? new Magnetometer(motionManager) : null;
```

Reply:
Fixed :-)